core-web/libs/edit-content/src/lib/fields/dot-edit-content-host-folder-field/store/host-folder-field.store.ts

```diff
@@ -0,0 +1,120 @@
+import { tapResponse } from '@ngrx/operators';
+import { signalStore, withState, withComputed, withMethods, patchState } from '@ngrx/signals';
+import { rxMethod } from '@ngrx/signals/rxjs-interop';
+import { of, pipe } from 'rxjs';
+
+import { computed, inject } from '@angular/core';
+
+import { exhaustMap, switchMap, tap, map, filter } from 'rxjs/operators';
+
+import {
+    CustomTreeNode,
+    TreeNodeItem,
+    TreeNodeSelectItem
+} from '../../../models/dot-edit-content-host-folder-field.interface';
+import { DotEditContentService } from '../../../services/dot-edit-content.service';
+
+type HostFolderFiledState = {
+    nodeSelected: TreeNodeItem | null;
+    nodeExpaned: TreeNodeSelectItem['node'] | null;
+    tree: TreeNodeItem[];
+    status: 'idle' | 'pending' | 'fulfilled' | { error: string };
```

Comment:
should we add another prop for the error? Looks ambiguous to have an error object inside a property that is mostly an string

Reply:
the message of error is not a status. Maybe have `status` (with 'error') and `error: string` in the store. 
We have centralized this constant
```
// core-web/libs/dotcms-models/src/lib/shared-models.ts
export enum ComponentStatus {
    INIT = 'INIT',
    LOADING = 'LOADING',
    LOADED = 'LOADED',
    SAVING = 'SAVING',
    IDLE = 'IDLE'
}
```

We could add `ERROR` and use from it.

Reply:
@oidacra @zJaaal done